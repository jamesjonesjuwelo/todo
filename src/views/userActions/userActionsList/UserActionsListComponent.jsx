import React, { Component } from 'react';
import { connect } from 'react-redux';
import { compose } from 'redux';
import styled from 'styled-components';
import { getUserActions } from '../../../store/action/userActionsAction';
import { toggleUserActionsOverlay } from '../../../store/action/commonAction';
import bindActionCreators from '../../../../util/bindActionCreators';
import UserActionComponent from '../userAction/UserActionComponent';
import IconComponent from '../../../component/icon/IconComponent';

export default class UserActionsListComponent extends Component {
  
  componentDidMount() {
    this.props.getUserActions();
  }
  
  closeUserActionsOverlay = () => {
    this.props.toggleUserActionsOverlay();
  };
  
  render() {
    const { userActions } = this.props;
    
    return (
      <Overlay>
        <Container>
          <IconComponent
            method={this.closeUserActionsOverlay}
            src="/icons/cross.png"
            styles={crossStyles}
          />
          <div className="user-actions-container">
            {userActions.length
              ? userActions.map(userAction =>
                <UserActionComponent key={`userAction-${userAction.id}`} userAction={userAction} />)
              : <NoItems>There are no recorded actions for this user.</NoItems>
            }
          </div>
        </Container>
      </Overlay>
    )
  }
}

const mapStateToProps = (state) => ({
  userActions: state.userActionsReducer.userActions
});

const mapDispatchToProps = (dispatch, props) => bindActionCreators({
  getUserActions,
  toggleUserActionsOverlay
}, dispatch, props);

const enhance = compose(
  connect(mapStateToProps, mapDispatchToProps)
);

const transition = `
    -moz-transition: all 0.15s ease-in;
    -o-transition: all 0.15s ease-in;
    -webkit-transition: all 0.15s ease-in;
    transition: all 0.15s ease-in;
`;

const Overlay = styled.div`
  position: fixed;
  left: 0;
  top: 0;
  height: 100vh;
  width: 100%;
  display: flex
  justify-content: center;
  align-items: center;
  background-color: rgba(16, 16, 16, 0.6);
`;

const Container = styled.div`
  width: 70%;
  height: 70%;
  padding: 20px;
  position: relative;
  background-color: white;
  overflow: scroll;
`;

const crossStyles = `
  position: absolute;
  right: 10px;
  top: 10px;
  width: 25px;
  height: 25px;
  cursor: pointer;
  opacity: 0.5;
  
  :hover {
    opacity: 1;
  }
`;

const NoItems = styled.p`
  text-align: center;
`;


export const UserActionsComponentRoute = enhance(UserActionsListComponent);