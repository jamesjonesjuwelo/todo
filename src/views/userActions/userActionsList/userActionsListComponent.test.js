import React from 'react';
import { shallow } from 'enzyme';
import UserActionsListComponent from './UserActionsListComponent';
import 'jest-styled-components';

const props = {
  userActions: [
    {
      id: 1,
      description: 'Test Description',
      type: 'Test type'
    },
    {
      id: 2,
      description: 'Test2 Description',
      type: 'Test2 type'
    }
  ],
  getUserActions: jest.fn(),
  toggleUserActionsOverlay: jest.fn()
  
};

describe('User Actions List Component', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow( <UserActionsListComponent {...props} /> );
  });
  
  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});