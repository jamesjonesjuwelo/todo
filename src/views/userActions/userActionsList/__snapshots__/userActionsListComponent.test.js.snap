// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Actions List Component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserActionsListComponent
    getUserActions={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    toggleUserActionsOverlay={[MockFunction]}
    userActions={
      Array [
        Object {
          "description": "Test Description",
          "id": 1,
          "type": "Test type",
        },
        Object {
          "description": "Test2 Description",
          "id": 2,
          "type": "Test2 type",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div>
        <IconComponent
          method={[Function]}
          src="/icons/cross.png"
          styles="
  position: absolute;
  right: 10px;
  top: 10px;
  width: 25px;
  height: 25px;
  cursor: pointer;
  opacity: 0.5;
  
  :hover {
    opacity: 1;
  }
"
        />
        <div
          className="user-actions-container"
        >
          <userActionComponent
            userAction={
              Object {
                "description": "Test Description",
                "id": 1,
                "type": "Test type",
              }
            }
          />
          <userActionComponent
            userAction={
              Object {
                "description": "Test2 Description",
                "id": 2,
                "type": "Test2 type",
              }
            }
          />
        </div>
      </styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <IconComponent
            method={[Function]}
            src="/icons/cross.png"
            styles="
  position: absolute;
  right: 10px;
  top: 10px;
  width: 25px;
  height: 25px;
  cursor: pointer;
  opacity: 0.5;
  
  :hover {
    opacity: 1;
  }
"
          />,
          <div
            className="user-actions-container"
          >
            <userActionComponent
              userAction={
                Object {
                  "description": "Test Description",
                  "id": 1,
                  "type": "Test type",
                }
              }
            />
            <userActionComponent
              userAction={
                Object {
                  "description": "Test2 Description",
                  "id": 2,
                  "type": "Test2 type",
                }
              }
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "method": [Function],
            "src": "/icons/cross.png",
            "styles": "
  position: absolute;
  right: 10px;
  top: 10px;
  width: 25px;
  height: 25px;
  cursor: pointer;
  opacity: 0.5;
  
  :hover {
    opacity: 1;
  }
",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <userActionComponent
                userAction={
                  Object {
                    "description": "Test Description",
                    "id": 1,
                    "type": "Test type",
                  }
                }
              />,
              <userActionComponent
                userAction={
                  Object {
                    "description": "Test2 Description",
                    "id": 2,
                    "type": "Test2 type",
                  }
                }
              />,
            ],
            "className": "user-actions-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "userAction-1",
              "nodeType": "function",
              "props": Object {
                "userAction": Object {
                  "description": "Test Description",
                  "id": 1,
                  "type": "Test type",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "userAction-2",
              "nodeType": "function",
              "props": Object {
                "userAction": Object {
                  "description": "Test2 Description",
                  "id": 2,
                  "type": "Test2 type",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div>
          <IconComponent
            method={[Function]}
            src="/icons/cross.png"
            styles="
  position: absolute;
  right: 10px;
  top: 10px;
  width: 25px;
  height: 25px;
  cursor: pointer;
  opacity: 0.5;
  
  :hover {
    opacity: 1;
  }
"
          />
          <div
            className="user-actions-container"
          >
            <userActionComponent
              userAction={
                Object {
                  "description": "Test Description",
                  "id": 1,
                  "type": "Test type",
                }
              }
            />
            <userActionComponent
              userAction={
                Object {
                  "description": "Test2 Description",
                  "id": 2,
                  "type": "Test2 type",
                }
              }
            />
          </div>
        </styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <IconComponent
              method={[Function]}
              src="/icons/cross.png"
              styles="
  position: absolute;
  right: 10px;
  top: 10px;
  width: 25px;
  height: 25px;
  cursor: pointer;
  opacity: 0.5;
  
  :hover {
    opacity: 1;
  }
"
            />,
            <div
              className="user-actions-container"
            >
              <userActionComponent
                userAction={
                  Object {
                    "description": "Test Description",
                    "id": 1,
                    "type": "Test type",
                  }
                }
              />
              <userActionComponent
                userAction={
                  Object {
                    "description": "Test2 Description",
                    "id": 2,
                    "type": "Test2 type",
                  }
                }
              />
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "method": [Function],
              "src": "/icons/cross.png",
              "styles": "
  position: absolute;
  right: 10px;
  top: 10px;
  width: 25px;
  height: 25px;
  cursor: pointer;
  opacity: 0.5;
  
  :hover {
    opacity: 1;
  }
",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <userActionComponent
                  userAction={
                    Object {
                      "description": "Test Description",
                      "id": 1,
                      "type": "Test type",
                    }
                  }
                />,
                <userActionComponent
                  userAction={
                    Object {
                      "description": "Test2 Description",
                      "id": 2,
                      "type": "Test2 type",
                    }
                  }
                />,
              ],
              "className": "user-actions-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "userAction-1",
                "nodeType": "function",
                "props": Object {
                  "userAction": Object {
                    "description": "Test Description",
                    "id": 1,
                    "type": "Test type",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "userAction-2",
                "nodeType": "function",
                "props": Object {
                  "userAction": Object {
                    "description": "Test2 Description",
                    "id": 2,
                    "type": "Test2 type",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
